File: backend-dotnet/SahayaSetu/DTO/DonationDto.cs
using SahayaSetu.Models;
using System.ComponentModel.DataAnnotations;

namespace SahayaSetu.DTO
{
    public class DonationDto
    {
        [Required]
        public decimal Amount { get; set; }

        [Required]
        public PaymentMode PaymentMode { get; set; }

        public string? PaymentReference { get; set; }
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/dtos/DonationDto.java
package com.sahayaSetu.dtos;

import com.sahayaSetu.entities.enums.PaymentMode;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class DonationDto {
    @NotNull(message = "Amount is required")
    @Min(value = 1, message = "Amount must be greater than 0")
    private BigDecimal amount;

    @NotNull(message = "Payment mode is required")
    private PaymentMode paymentMode;

    private String paymentReference; // Optional transaction ID
}
